agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/components/event/photo/EventDetailPhotoHolder.kt

```diff
@@ -0,0 +1,136 @@
+package com.pronoidsoftware.agenda.presentation.detail.components.event.photo
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.pronoidsoftware.agenda.presentation.R
+import com.pronoidsoftware.core.presentation.designsystem.PlusIcon
+import com.pronoidsoftware.core.presentation.designsystem.TaskyLightBlue2
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+
+@Composable
+private fun PhotoBorder(
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    color: Color = TaskyLightBlue2,
+    size: Dp = 60.dp,
+    strokeWidth: Dp = 2.dp,
+    cornerRadius: Dp = 5.dp,
+    content: @Composable () -> Unit = { },
+) {
+    val density = LocalDensity.current
+    val cornerRadiusPx = with(density) { cornerRadius.toPx() }
+    Box(
+        modifier = modifier
+            .size(size + strokeWidth)
+            .clip(RoundedCornerShape(cornerRadius))
+            .clickable {
+                onClick()
+            },
+        contentAlignment = Alignment.Center,
+    ) {
+        content()
+        Canvas(
+            modifier = Modifier.fillMaxSize(),
+        ) {
+            drawRoundRect(
+                color = color,
+                cornerRadius = CornerRadius(cornerRadiusPx, cornerRadiusPx),
+                style = Stroke(
+                    width = with(density) { strokeWidth.toPx() },
+                ),
+            )
+        }
```

Comment:
Instead of a custom canvas, you could also simply use the `drawBehind` modifier for the box here which also gives you a reference to a canvas which draws behind the Box content

Reply:
Ah, but here I want the canvas on top of the content within the box to give it a border. However, this made me realize there's a border modifier, thanks!

Reply:
True, I didn't notice this is just for the border 😄 
But there's also a modifier to draw something on top of the content:
```
Modifier.drawWithContent {
    drawContent() // This draws the content
    
    drawRect(Color.Red) // This is now drawn on top
}
```